src/components/App.js

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useState } from "react";
+import { Routes, Route, Link } from "react-router-dom";
+import { auth } from "../firebase";
+import { onAuthStateChanged } from "firebase/auth";
+import logo from "../logo.png";
+import "./App.css";
+import Composer from "./Composer.js";
+import NewsFeed from "./NewsFeed.js";
+import AuthForm from "./AuthForm.js";
+
+const App = () => {
+  const [loggedInUser, setLoggedInUser] = useState();
+
+  useEffect(() => {
+    onAuthStateChanged(auth, (user) => {
+      if (user) {
+        // User is signed in, saved logged-in user to state
+        setLoggedInUser(user);
+        return;
+      }
+
+      // User is signed out
+      setLoggedInUser(null);
+    });
+  }, []);
+
+  const authForm = <AuthForm />;
```

Comment:
This seems a bit redundant to me. Why not just use the Component in the element attribute for your route? And if we use a variable, we could do so outside of this component as well.